src/enum_net.cpp

```diff
@@ -168,19 +168,24 @@ namespace {
 
 #if TORRENT_USE_NETLINK
 
-	int read_nl_sock(int sock, span<char> buf, std::uint32_t const seq, std::uint32_t const pid)
+	int read_nl_sock(int sock, std::vector<char>& buf, std::uint32_t const seq, std::uint32_t const pid)
 	{
 		nlmsghdr* nl_hdr;
 
-		int msg_len = 0;
-
+		int increment = 1024;
+		int const max_increment = 8192;
+		int read_cursor = 0;
 		for (;;)
 		{
-			auto next_msg = buf.subspan(msg_len);
-			int const read_len = int(recv(sock, next_msg.data(), static_cast<std::size_t>(next_msg.size()), 0));
+			if (read_cursor + (increment / 2) > int(buf.size()))
```

Comment:
This assumes messages will never be larger than 512 bytes, which is probably fine in this case, but if you want to be really pedantic netlink messages can be up to 32KB on current kernels.

Reply:
good point. when I started this patch I was thinking netlink sockets were stream-like, but I realize they aren't.

Reply:
also, it seems a bit unnecessary to complicate this by making the headroom be `increment / 2`, instead of just a fixed number. The minimum number will be the bottleneck anyway